src/Microsoft.VisualStudio.Services.Agent/Util/IOUtil.cs

```diff
@@ -151,6 +151,12 @@ public static string GetWorkPath(IHostContext hostContext)
                 settings.WorkFolder);
         }
 
+        public static string GetHomePath()
+        {
+            var envHome = Constants.Agent.Platform == Constants.OSPlatform.Windows ? "HOMEPATH" : "HOME";
```

Comment:
You might want to prefer USERPROFILE over HOMEPATH/HOMEDRIVE (note, homepath does not include the drive letter, homedrive).

Based on 10 minute internet search, it looks like USERPROFILE is the drive where the local user (Windows) profile is. It's always local. Whereas network admins sometimes create a mapped network drive, and repoint homedrive/homepath to the mapped network drive (to store pictures, documents, etc).